src/Microsoft.AspNet.Mvc.Core/Logging/DefaultActionSelectorLoggerExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Mvc.ActionConstraints;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Mvc.Logging
+{
+    public static class DefaultActionSelectorLoggerExtensions
+    {
+        private static Action<ILogger, string, Exception> _ambiguousActions;
+        private static Action<ILogger, string, string, IActionConstraint, Exception> _constraintMismatch;
+
+        static DefaultActionSelectorLoggerExtensions()
+        {
+            _ambiguousActions = LoggerMessage.Define<string>(
+                LogLevel.Error,
+                1,
```

Comment:
Yeah, I'm with @rynowak on this one, everything except for eventId is pretty self-explanatory.